File: frontend/src/services/environment.ts
import api from '../config/api';

export interface Environment {
  id: number;
  name: string;
  display_name: string;
  color: string;
  sort_order: number;
  description: string;
  created_at: string;
  updated_at: string;
}

export interface CreateEnvironmentRequest {
  name: string;
  display_name: string;
  color?: string;
  sort_order?: number;
  description?: string;
}

export interface UpdateEnvironmentRequest {
  name?: string;
  display_name?: string;
  color?: string;
  sort_order?: number;
  description?: string;
}

export const environmentService = {
  list: async (page: number = 1, pageSize: number = 100) => {
    const response = await api.get<{
      environments: Environment[];
      total: number;
      page: number;
      page_size: number;
    }>('/environments', { params: { page, page_size: pageSize } });
    return response.data;
  },

  get: async (id: number) => {
    const response = await api.get<{ environment: Environment }>(`/environments/${id}`);
    return response.data;
  },

  create: async (data: CreateEnvironmentRequest) => {
    const response = await api.post<{ environment: Environment }>('/environments', data);
    return response.data;
  },

  update: async (id: number, data: UpdateEnvironmentRequest) => {
    const response = await api.put<{ environment: Environment }>(`/environments/${id}`, data);
    return response.data;
  },

  delete: async (id: number) => {
    const response = await api.delete(`/environments/${id}`);
    return response.data;
  },
};

File: backend/internal/models/environment.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Environment 环境模型
type Environment struct {
	ID          uint64         `gorm:"primaryKey" json:"id"`
	Name        string         `gorm:"size:50;not null;uniqueIndex" json:"name"`
	DisplayName string         `gorm:"size:100;not null" json:"display_name"`
	Color       string         `gorm:"size:20" json:"color"`
	SortOrder   int            `gorm:"default:0" json:"sort_order"`
	Description string         `json:"description"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

// TableName 指定表名
func (Environment) TableName() string {
	return "environments"
}
